### PR TITLE
[Feat] New talent request status UI

### LINKED ISSUE #16826: ✨ Frontend: Update talent request statuses
### Details

Update the frontend for the new status details for a talent request.

### Design file & designer

 - Designer: @Jerryescandon
 - [Figma link](https://www.figma.com/design/OcJlqTARk7UEG2aK7H0AG5/Talent-requests--Communities-?node-id=146-18360&p=f&t=GlNSm9khWxWVC8qj-0)

### 📸 Screenshots

Image

Image

Image

Image

### 🌎 Localization

_No response_

### Proposed Implementation

 - Update `SearchRequestTable` to use new `status` and `details` info
 - Update sidebar to include new status details (SEE: [Applicaiton sidebar](https://github.com/GCTC-NTGC/gc-digital-talent/tree/main/apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/Sidebar) for inspiration

 

### ✅ Acceptance Criteria

- [ ] Search request updated to use new status
- [ ] Sidebar updated with new status details
- [ ] Requests table shows new status and status details columns, replacing old status column
- [ ] New status column is sortable
- [ ] New status and status details column are filterable

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #16826 

## 👋 Introduction

Updates talent requests to use the new status with details.

## 🧪 Testing

1. Build `pnpm dev:fresh`
2. Login as `admin@test.com`
3. Navigate to talent requests `/admin/talent-requests`
4. Confirm the new status and details columns
5. Confirm you can filter and sort statuses
6. Navigate to a specific request
7. Confirm new status and details appear in the sidebar
8. Confirm you can update and get the expected values back

## 📸 Screenshot

<details><summary>Table</summary>
<p>

swappy-20260522_083509

</p>
</details> 

<details><summary>New</summary>
<p>

swappy-20260522_083520
swappy-20260522_083533

</p>
</details> 

<details><summary>In progress</summary>
<p>

swappy-20260522_083541
swappy-20260522_083603

</p>
</details> 

<details><summary>Closed</summary>
<p>

swappy-20260522_083612
swappy-20260522_083618

</p>
</details>

### CODE REVIEW

File: api/app/GraphQL/Validators/Mutation/UpdatePoolCandidateSearchRequestStatusValidator.php [hunk 1]
```diff
@@ -22,10 +22,12 @@ public function rules(): array
                 ]),
             ],
             'poolCandidateSearchRequest.inProgressDetails' => [
+                'nullable',
                 'required_if:poolCandidateSearchRequest.status,'.TalentRequestStatus::IN_PROGRESS->name,
                 Rule::in(array_column(TalentRequestInProgressDetail::cases(), 'name')),
             ],
             'poolCandidateSearchRequest.closedDetails' => [
+                'nullable',
```
Comment (tristan-orourke) [inline api/app/GraphQL/Validators/Mutation/UpdatePoolCandidateSearchRequestStatusValidator.php:30]: How does nullable interact with required_if?

Did you have to add this for the UI to work properly? If that's the case, I'm surprised it wasn't caught by unit tests. Is there a case we're missing?

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+import talentRequestMessages from "~/messages/talentRequestMessages";
+
+const COLOUR_MAP: Record<
+  TalentRequestStatus | "default",
+  StatusButtonProps["color"]
+> = {
+  [TalentRequestStatus.New]: "warning",
+  [TalentRequestStatus.InProgress]: "primary",
+  [TalentRequestStatus.Closed]: "black",
```
Comment (tristan-orourke) [inline apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:41]: Closed colour was gray elsewhere, is that intended?

File: api/app/GraphQL/Validators/Mutation/UpdatePoolCandidateSearchRequestStatusValidator.php (hunk 1 above)
Comment (esizer) [reply api/app/GraphQL/Validators/Mutation/UpdatePoolCandidateSearchRequestStatusValidator.php:30]: It first allows `null` values then, if the condition in `required_id` are met, it enforces that the value is not `null`  or empty. The `nullable` rule does not override the `required_if` part.

Added more tests for this in [add test cases for allowing null](https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857/commits/023443ebee4991935ea172f59b6410037f0ad26e)

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+                )}
+                {currentStatus === TalentRequestStatus.Closed && (
+                  <Select
+                    id="closedDetails"
+                    name="closedDetails"
+                    label={intl.formatMessage({
+                      defaultMessage: "Closed details",
+                      id: "xxiQ9n",
+                      description:
+                        "Label for in-progress talent request details",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx]: Colmpleted manually for ID generation [fix closed details description](https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857/commits/0b7e05c5181486b4bd95d5698faf775082db081c)

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+        : undefined,
+    },
+  });
+  const currentStatus = methods.watch("talentRequestStatus");
+  const notAvailable = intl.formatMessage(commonMessages.notAvailable);
+
+  const handleSubmit = async (values: FormValues) => {
+    if (fetching) return;
+
+    await executeMutation({
```
Comment (tristan-orourke) [inline apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:157]: **Stale detail fields sent on submit**

`handleSubmit` always sends `inProgressDetails`, `closedDetails`, and `followUpDate` regardless of the selected status. Even though `handleFormChange` clears the non-active fields when the radio changes, there are edge cases (e.g. toggling between statuses) where a previously-selected value could persist in form state.

Suggestion: null out irrelevant fields at submit time based on the selected status, so the API always gets a clean payload:

```ts
input: {
  status: values.talentRequestStatus,
  inProgressDetails: values.talentRequestStatus === TalentRequestStatus.InProgress
    ? values.inProgressDetails : null,
  closedDetails: values.talentRequestStatus === TalentRequestStatus.Closed
    ? values.closedDetails : null,
  followUpDate: values.talentRequestStatus === TalentRequestStatus.InProgress
    ? values.followUpDate : null,
},
```

-- generated by Claude Code

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+      .then((res) => {
+        if (
+          res.error ||
+          !res.data?.updatePoolCandidateSearchRequestStatus?.id
+        ) {
+          throw new Error(res.error?.toString() ?? "Unknown error");
+        }
+
+        toast.success(
+          intl.formatMessage({
```
Comment (tristan-orourke) [inline apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:184]: **Missing form reset after successful submit**

After a successful mutation the dialog closes, but `methods.reset()` is never called. If the user reopens the dialog, the form will show stale `defaultValues` from the initial render rather than the newly saved values (the old component called `methods.resetField` on success).

Consider calling `methods.reset(values)` on success here, or confirming that the fragment cache refetch causes a remount that resets the form.

-- generated by Claude Code

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx (hunk 2 above)
Comment (esizer) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:41]: Black is actually gray. It is confusing but designers refer to this as black in their designs so it was chosen for the underlyuing component to match designs. So while the colour is gray, the term we use to refer to it is black :sweat_smile: 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/1285a0c666745837b1274e274a149d7f6be1fe1f/packages/ui/src/components/Button/StatusButton.tsx#L54-L59

Comment (tristan-orourke) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:41]: That's fine, but why is it inconsistent with https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857/changes#diff-18e4d42ef635cba29b3dc8312c765e5de0f9fa165c957a8166eb5f520270c6f5R151

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx (hunk 4 above)
Comment (esizer) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:157]: Done in [null unused fields during mutation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857/commits/f2d0bdd4b2220c5c179869c864d233111ea167f6)

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx (hunk 5 above)
Comment (esizer) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:184]: Fixed in [reset form on success](https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857/commits/f5eae7cdd4a1b9241d3ebc6d846eef7994ba2b5a)

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+                      }}
+                      options={narrowEnumType(
+                        unpackMaybes(options?.inProgressDetails),
+                        "TalentRequestInProgressDetail",
+                      ).map((detail) => ({
+                        value: detail.value,
+                        label: detail.label.localized ?? notAvailable,
+                      }))}
+                    />
+                    <DateInput
```
Comment (tristan-orourke) [inline apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:296]: When opening the dialog for a request with an existing followup date, the date input appears to be pre-populated, but the form submit handler seems to see at as empty until the value is changed for each input element.

Image

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx (hunk 2 above)
Comment (esizer) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:41]: Because our component library is all out of sync and not consistent :sweat_smile: These are the kinds of things we woulkd like to address with the design system :stuck_out_tongue:

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx (hunk 6 above)
Comment (esizer) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:296]: That was already fixed in [fix default form date](https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857/commits/e920a0117db0eb27db9b941cfb285a30e6f5823e)

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx (hunk 2 above)
Comment (tristan-orourke) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/TalentRequestStatusDialog.tsx:41]: Right...

**Comment by tristan-orourke** [comment]
I'd love to add Status Details to the filters, I feel like it would be useful. However, I know it would be a bit more complicated than usual because its actually two fields. How hard would it be to build a multiselect which represents a union of In Progress Details enums and Closed Details enums, and then sort those into filters properly before sending the query?

Or maybe even just having two fields in the filter dialog?

image

**Review by tristan-orourke — APPROVED** [review]
You know what, I'll approve this as is, lets consider whether my previous [comment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/16857#issuecomment-4521238246) about filters should be turned into another ticket.
